Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +124,31 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (array != null && index >= 0 && index <= array.Length - 1)
+            {
+                int[] newArr = new int[array.Length + 1];
+                newArr[index] = number;
+                for (int i = 0; i < array.Length; ++i)
+                {
+                    if (i < index)
+                    {
+                        newArr[i] = array[i];
+                    }
+                    else {
+                        newArr[i + 1] = array[i];
+                    }
+                }
+                return newArr;
+            }
+            if (index < 0 || index > array.Length - 1)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4940124)